freqtrade/optimize/hyperopt.py

```diff
@@ -342,9 +348,25 @@ def get_optimizer(self, dimensions, cpu_count) -> Optimizer:
             random_state=self.config.get('hyperopt_random_state', None)
         )
 
-    def run_optimizer_parallel(self, parallel, asked) -> List:
+    def fix_optimizer_models_list(self):
+        """
+        WORKAROUND: Since skopt is not actively supported, this resolves problems with skopt
+        memory usage, see also: https://github.com/scikit-optimize/scikit-optimize/pull/746
+
+        This may cease working when skopt updates if implementation of this intrinsic
+        part changes.
+        """
+        n = len(self.opt.models) - SKOPT_MODELS_MAX_NUM
+        # Keep no more than 2*SKOPT_MODELS_MAX_NUM models in the skopt models list,
+        # remove the old ones. These are no really needed, the current model
+        # from the estimator is only used.
```

Comment:
"is the only one used"? 

maybe from upstream ... ?

Reply:
Done, rephrased.

What's "from upstream"?

Reply:
well my assumption was that part of this solution is also part of the PR against skopt ... which in this case would be the "upstream" project for this workaround.

Reply:
no, I borrowed the main idea only, the implementation is dufferent to suite our needs.

To help reviewing: the results (this PR and the develop) can be compared with same `--random-state NNN`, should be same (checked this at my site during preparation of this solution). It proves (besides skopt code review) that history of models is not used in skopt for optimization.

Reply:
i almost always test hyperopt with `--random-state` (at least when i depend on the results)  ... that's the only way to have reliable results i think...